packages/typescript-checker/src/typescript-checker.ts

```diff
@@ -1,201 +1,144 @@
-            this.fs.watchFile(filePath, callback);
-            return {
-              close: () => {
-                delete this.fs.getFile(filePath)!.watcher;
-              },
-            };
-          },
-          writeFile: (filePath, data) => {
-            this.fs.writeFile(filePath, data);
-          },
-          createDirectory: () => {
-            // Idle, no need to create directories in the hybrid fs
-          },
-          clearScreen() {
-            // idle, never clear the screen
-          },
-          getModifiedTime: (fileName) => {
-            return this.fs.getFile(fileName)!.modifiedTime;
-          },
-          watchDirectory: (): ts.FileWatcher => {
-            // this is used to see if new files are added to a directory. Can safely be ignored for mutation testing.
-            return {
-              // eslint-disable-next-line @typescript-eslint/no-empty-function
-              close() {},
-            };
-          },
-        },
-        undefined,
-        (error) => this.currentErrors.push(error),
-        (status) => this.logDiagnostic('status')(status),
-        (summary) => {
-          this.logDiagnostic('summary')(summary);
-          summary.code !== FILE_CHANGE_DETECTED_DIAGNOSTIC_CODE && this.resolveCheckResult();
-        }
-      ),
-      [this.tsconfigFile],
-      {}
-    );
-    compiler.build();
-    const result = await this.currentTask.promise;
-    if (result.status === CheckStatus.CompileError) {
-      throw new Error(`TypeScript error(s) found in dry run compilation: ${result.reason}`);
+    const { dependencyFiles, errors } = await this.tsCompiler.init();
+
+    if (errors.length) {
+      throw new Error(`TypeScript error(s) found in dry run compilation: ${this.formatErrors(errors)}`);
     }
-  }
 
-  private guardTSConfigFileExists() {
-    if (!ts.sys.fileExists(this.tsconfigFile)) {
-      throw new Error(
-        `The tsconfig file does not exist at: "${path.resolve(
-          this.tsconfigFile
-        )}". Please configure the tsconfig file in your stryker.conf file using "${propertyPath<StrykerOptions>('tsconfigFile')}"`
-      );
+    if (Object.keys(dependencyFiles).length === 0) {
+      throw new Error('No sourcefiles were loaded in the compiler.');
     }
+
+    this.sourceFiles = dependencyFiles;
   }
 
-  /**
-   * Checks whether or not a mutant results in a compile error.
-   * Will simply pass through if the file mutated isn't part of the typescript project
-   * @param mutant The mutant to check
-   */
-  public async check(mutant: Mutant): Promise<CheckResult> {
-    if (this.fs.existsInMemory(mutant.fileName)) {
-      this.clearCheckState();
-      this.fs.mutate(mutant);
-      return this.currentTask.promise;
-    } else {
-      // We allow people to mutate files that are not included in this ts project
+  public async check(mutants: MutantTestCoverage[]): Promise<Array<{ mutant: MutantTestCoverage; checkResult: CheckResult }>> {
```

Comment:
Probably we can split this into an `optimisticCheck` (check the entire group) and `pessimisticCheck` (check mutants one by one) method?